Sources/_StringProcessing/ConsumerInterface.swift

```diff
@@ -77,10 +72,10 @@ extension AST.Atom {
   }
 
   func generateConsumer(
-    _ opts: CharacterClass.MatchLevel
+    _ opts: MatchingOptions
   ) throws -> Program<String>.ConsumeFunction? {
     // TODO: Wean ourselves off of this type...
-    if let cc = self.characterClass?.withMatchLevel(opts) {
+    if let cc = self.characterClass?.withMatchLevel(opts.matchLevel) {
```

Comment:
(post this PR) definitely be thinking about how we should model character classes better, given that options can change their definitions

Reply:
Yeah, we at least need to transition from `CharacterClass.MatchLevel` to something like the new `MatchingOptions` type.